### DISCUSSION TITLE
Now Playing Menu doesnt stay inside the collection?

### DISCUSSION THREAD

**Discussion by JuLi0n21** [ppy/osu#24025] [Q&A]
Are there any plans to make the Now playing menu stick to the playlist choosen? it feels realy unintuitve that after choosing a collection u cant use the buttons to iterate through it.

Expecteted behavoir:

https://github.com/ppy/osu/assets/97122234/5947e6be-5d94-429d-ab2f-7d64810cb08f

Acutal behavoir: 

https://github.com/ppy/osu/assets/97122234/ed587959-e3e8-40ad-a599-fcbd1d45eb62

**Comment by peppy** [comment]
This would require `MusicController` to be aware of the currently selected collection. But if we're going to do this, maybe it should also be a global bindable which is applied to song selection's view as well?

**Comment by JuLi0n21** [comment]
I think making the MusicController aware is a good idea, if it were to be passed to the Musiccontroller and used in the filtering for the realm query it should in theory also be applied directly to the SongSelect and CollectionDropdown.

i wasnt able to find out exactly how the SongSelect handels the Collection Filtering but it looked like the way its done in the CollectionDropdown is not how it should be keept (Horrible performance in comparison to Songselect)
